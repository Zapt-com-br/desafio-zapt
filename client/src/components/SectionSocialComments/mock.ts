export default [
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'Eu disse que seria a melhor pessoa para ajudar a criar uma ambiente foda, para construir coisas incríveis'
    }
  },
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'Depois que entrei pra Zapt consegui ter clareza sobre o futuro da minha carreira'
    }
  },
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'A oportunidade de ter um mentor que é especialista frontend é um passo gigante pra acelerar a carreira'
    }
  },
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'Ceis também querem um caneca da Zapt? Tem que ser ousado pra conseguir em! hahaha'
    }
  },
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'A oportunidade de ter um mentor que é especialista frontend é um passo gigante pra acelerar a carreira'
    }
  },
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'Depois que entrei pra Zapt consegui ter clareza sobre o futuro da minha carreira'
    }
  },
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'A oportunidade de ter um mentor que é especialista frontend é um passo gigante pra acelerar a carreira'
    }
  },
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'A oportunidade de ter um mentor que é especialista frontend é um passo gigante pra acelerar a carreira'
    }
  },
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'Ceis também querem um caneca da Zapt? Tem que ser ousado pra conseguir em! hahaha'
    }
  }
]
