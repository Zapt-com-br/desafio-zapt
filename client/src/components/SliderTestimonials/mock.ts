const user = {
  name: 'Alan Gabriel',
  avatar_url: '/img/profile.jpg',
  address: {
    city: 'Arinos',
    state: 'Minas Gerais'
  }
}

export default [
  {
    testimonial: {
      text:
        'Eu disse que seria a melhor pessoa para ajudar a criar uma ambiente foda, para construir coisas incríveis',
      thumbnail: '/img/IMG_1764.jpg'
    },
    user
  },
  {
    testimonial: {
      text:
        'A oportunidade de ter um mentor que é especialista frontend é um passo gigante pra acelerar a carreira',
      thumbnail: '/img/IMG_1756.jpg'
    },
    user
  },
  {
    testimonial: {
      text:
        'Depois que entrei pra Zapt consegui ter clareza sobre o futuro da minha carreira',
      thumbnail: '/img/IMG_1758.jpg'
    },
    user
  }
]
