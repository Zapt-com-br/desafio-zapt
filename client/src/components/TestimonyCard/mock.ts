import thumbnail from '../../../public/img/IMG_1756.jpg'
import avatar_url from '../../../public/img/profile.jpg'

export default {
  testimonial: {
    text:
      'Depois que entrei pra Zapt consegui ter clareza sobre o futuro da minha carreira',
    thumbnail
  },
  user: {
    name: 'Alan Gabriel',
    avatar_url,
    address: {
      city: 'Arinos',
      state: 'Minas Gerais'
    }
  }
}
