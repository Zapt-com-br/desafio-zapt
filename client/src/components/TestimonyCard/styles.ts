import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: relative;

    width: 100%;
    max-width: 58.8rem;
    height: 40rem;
    border-radius: ${theme.border.radius};
    background: ${theme.colors.drWhite};
    overflow: hidden;
    transition: all 0.3s;

    ::before {
      position: absolute;
      content: '';
      width: 100%;
      height: 100%;
      top: 0;
      left: 0;
      background: radial-gradient(
          112.99% 451.94% at 0% 100%,
          rgba(0, 0, 0, 0.7) 0%,
          rgba(0, 0, 0, 0) 100%
        ),
        url('/img/noise-overlay.png');

      z-index: ${theme.layers.base};
    }

    :hover {
      transform: scale(1.005);
    }

    :hover ${WrapperTestimonial} > span {
      transform: scale(1.02);
    }

    @media (max-width: 450px) {
      height: 30rem;
    }
  `}
`

export const Thumbnail = styled.div`
  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
    z-index: 0;
  }
`

export const WrapperTestimonial = styled.div`
  ${({ theme }) => css`
    position: absolute;
    bottom: 0;
    left: 0;
    z-index: ${theme.layers.alwaysOnTop};
    color: ${theme.colors.white};

    padding: ${theme.spacings.small};

    max-width: 460px;

    > span {
      display: block;
      font-size: ${theme.font.sizes.large};
      line-height: calc(${theme.spacings.small} + 0.2rem);

      letter-spacing: -0.01em;

      transition: all 0.3s;

      ::after {
        content: '”';
      }

      ::before {
        content: '“';
      }
    }
  `}
`

export const WrapperUser = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-areas: 'avatar name' 'avatar city';

    grid-template-columns: 40px auto;
    column-gap: ${theme.spacings.xsmall};

    margin-top: ${theme.spacings.xsmall};

    color: ${theme.colors.white};

    p {
      grid-area: name;

      font-size: ${theme.font.sizes.medium};
      letter-spacing: -0.01em;

      color: currentColor;
    }

    span {
      grid-area: city;

      font-size: ${theme.font.sizes.xsmall};
      opacity: 0.8;
      letter-spacing: -0.01em;
    }
  `}
`

export const Avatar = styled.div`
  ${({ theme }) => css`
    grid-area: avatar;
    width: 40px;
    height: 40px;
    object-fit: cover;
    border-radius: 100%;
    overflow: hidden;

    border: 1.5px solid ${theme.colors.white};

    img {
      margin-left: -2px !important;
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  `}
`
