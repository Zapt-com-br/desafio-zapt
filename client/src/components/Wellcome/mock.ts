export default {
  user: 'Marcio',
  content:
    'Para você aproveitar nossas oportunidades primeiro você precisa estar dentro de um grupo, estamos aqui para lhe ajudar. ',
  textCta: 'O que você prefere?',
  infoContact:
    'Qualquer dúvida que você tiver é só nos chamar no nosso número de WhatsApp.',
  numberPhone: '(99)12345-6789'
}
