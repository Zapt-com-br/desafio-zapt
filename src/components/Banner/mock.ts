export default {
  title: 'Bem vinda Polyane,',
  contactNumber: '11910898780'
}
