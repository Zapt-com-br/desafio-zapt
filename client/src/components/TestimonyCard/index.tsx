import * as S from './styles'

type TestimonialProps = {
  thumbnail: string
  text: string
}

type UserType = {
  name: string
  avatar_url: string
  address: {
    city: string
    state: string
  }
}

export type TestimonyCardProps = {
  testimonial: TestimonialProps
  user: UserType
}

const TestimonyCard = ({ testimonial, user }: TestimonyCardProps) => (
  <S.Wrapper>
    <S.WrapperTestimonial>
      <span>{testimonial.text}</span>
      <S.WrapperUser>
        <img src={user.avatar_url} alt={user.name} />
        <p>{user.name}</p>
        <span>
          {user.address.city} &bull; {user.address.state}
        </span>
      </S.WrapperUser>
    </S.WrapperTestimonial>
    <S.Thumbnail
      src={testimonial.thumbnail}
      alt={`Imagem de capa do depoimento de ${user.name}`}
    />
  </S.Wrapper>
)

export default TestimonyCard
