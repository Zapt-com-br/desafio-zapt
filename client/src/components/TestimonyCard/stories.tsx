import { Story, Meta } from '@storybook/react'
import TestimonyCard, { TestimonyCardProps } from '.'

export default {
  title: 'TestimonyCard',
  component: TestimonyCard
} as Meta

export const Default: Story<TestimonyCardProps> = (args) => (
  <TestimonyCard {...args} />
)

Default.args = {
  user: {
    name: 'Alan Gabriel',
    avatar_url: '/img/profile.jpg',
    address: {
      city: 'Arinos',
      state: 'Minas Gerias'
    }
  },
  testimonial: {
    text:
      'Depois que entrei pra Zapt consegui ter clareza sobre o futuro da minha carreira',
    thumbnail: '/img/IMG_1756.jpg'
  }
}
