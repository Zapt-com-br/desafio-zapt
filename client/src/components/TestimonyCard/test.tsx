import { screen } from '@testing-library/react'
import renderWithTheme from 'utils/tests/test-util'

import mock from 'components/TestimonyCard/mock'
import TestimonyCard from '.'

describe('<TestimonyCard />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<TestimonyCard {...mock} />)

    expect(screen.getByText(mock.testimonial.text)).toBeInTheDocument()

    expect(screen.getByText(mock.user.name)).toBeInTheDocument()

    expect(
      screen.getByText(`${mock.user.address.city} â€¢ ${mock.user.address.state}`)
    ).toBeInTheDocument()

    expect(screen.getByRole('img', { name: mock.user.name })).toHaveAttribute(
      'src',
      mock.user.avatar_url
    )

    expect(
      screen.getByRole('img', {
        name: `Imagem de capa do depoimento de ${mock.user.name}`
      })
    ).toHaveAttribute('src', mock.testimonial.thumbnail)

    expect(container.firstChild).toMatchSnapshot()
  })
})
