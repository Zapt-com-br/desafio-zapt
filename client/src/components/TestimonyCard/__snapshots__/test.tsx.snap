// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestimonyCard /> should render correctly 1`] = `
.c0 {
  position: relative;
  width: 100%;
  max-width: 58.8rem;
  height: 40rem;
  border-radius: 0.4rem;
  background: #FAFAFA;
  overflow: hidden;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0::before {
  position: absolute;
  content: '';
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  background: radial-gradient( 112.99% 451.94% at 0% 100%,rgba(0,0,0,0.7) 0%,rgba(0,0,0,0) 100% ),url('/img/noise-overlay.png');
  z-index: 10;
}

.c0:hover {
  -webkit-transform: scale(1.005);
  -ms-transform: scale(1.005);
  transform: scale(1.005);
}

.c0:hover .c1 > span {
  -webkit-transform: scale(1.02);
  -ms-transform: scale(1.02);
  transform: scale(1.02);
}

.c4 {
  width: 100%;
  height: 100%;
  object-fit: cover;
  z-index: 0;
}

.c2 {
  position: absolute;
  bottom: 0;
  left: 0;
  z-index: 50;
  color: #FFFFFF;
  padding: 2.4rem;
  max-width: 460px;
}

.c2 > span {
  display: block;
  font-size: 2.0rem;
  line-height: calc(2.4rem + 0.2rem);
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c2 > span::after {
  content: '”';
}

.c2 > span::before {
  content: '“';
}

.c3 {
  display: grid;
  grid-template-areas: 'avatar name' 'avatar city';
  grid-template-columns: 40px auto;
  -webkit-column-gap: 1.6rem;
  column-gap: 1.6rem;
  margin-top: 1.6rem;
  color: #FFFFFF;
}

.c3 img {
  grid-area: avatar;
  width: 40px;
  height: 40px;
  object-fit: cover;
  border-radius: 100%;
  border: 1.5px solid #FFFFFF;
}

.c3 p {
  grid-area: name;
  font-size: 1.6rem;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: currentColor;
}

.c3 span {
  grid-area: city;
  font-size: 1.2rem;
  opacity: 0.8;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
}

@media (max-width:450px) {
  .c0 {
    height: 30rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  >
    <span>
      Depois que entrei pra Zapt consegui ter clareza sobre o futuro da minha carreira
    </span>
    <div
      class="c3"
    >
      <img
        alt="Alan Gabriel"
        src="/img/profile.jpg"
      />
      <p>
        Alan Gabriel
      </p>
      <span>
        Arinos
         • 
        Minas Gerais
      </span>
    </div>
  </div>
  <img
    alt="Imagem de capa do depoimento de Alan Gabriel"
    class="c4"
    src="/img/IMG_1756.jpg"
  />
</div>
`
